src/app/modules/submission/encrypt-submission/encrypt-submission.service.ts

```diff
@@ -786,3 +787,67 @@ export const triggerVirusScanning = (
     return errAsync(new VirusScanFailedError())
   })
 }
+
+/**
+ * Downloads file from clean bucket
+ * @param cleanFileKey object key of the file in the clean bucket
+ * @param versionId id for versioning of the file in the clean bucket
+ * @returns okAsync(buffer) if file has been successfully downloaded from the clean bucket
+ * @returns errAsync(DownloadCleanFileFailedError) if file download failed
+ */
+export const downloadCleanFile = (cleanFileKey: string, versionId: string) => {
+  const logMeta = {
+    action: 'downloadCleanFile',
+    cleanFileKey,
+    versionId,
+  }
+
+  if (!validate(cleanFileKey)) {
+    logger.error({
+      message: 'Invalid clean file key - not a valid uuid',
+      meta: logMeta,
+    })
+
+    return errAsync(new InvalidFileKeyError())
+  }
+
+  let buffer = Buffer.alloc(0)
+
+  const writeStream = new Writable({
+    write(chunk, encoding, callback) {
```

Comment:
/nit can declare _encoding, since it is not used

Reply:
Done in [c77b2f3](https://github.com/opengovsg/FormSG/pull/6747/commits/c77b2f3bfbeed6fbfb3043374511515dff927fb2)